edit/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+use std::ffi::OsString;
+use std::str::FromStr;
+
+use radicle_common::args::{Args, Error, Help};
+use radicle_common::{keys, profile};
+use radicle_terminal as term;
+
+use librad::git::identities::{any, person, project, SomeIdentity};
+use librad::git::Urn;
+
+use link_identities::payload::{PersonPayload, ProjectPayload};
+
+use anyhow::anyhow;
+
+pub const HELP: Help = Help {
+    name: "edit",
+    description: env!("CARGO_PKG_DESCRIPTION"),
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad edit [<urn>] [<option>...]
+
+    Edits the identity document pointed to by the URN. If it isn't specified,
+    the current project is edited.
+
+Options
+
+    --help              Print help
+"#,
+};
+
+#[derive(Default, Debug, Eq, PartialEq)]
+pub struct Options {
+    pub urn: Option<Urn>,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut urn: Option<Urn> = None;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Value(val) if urn.is_none() => {
+                    let val = val.to_string_lossy();
+
+                    if let Ok(val) = Urn::from_str(&val) {
+                        urn = Some(val);
+                    } else {
+                        return Err(anyhow!("invalid URN '{}'", val));
+                    }
+                }
+                _ => return Err(anyhow::anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((Options { urn }, vec![]))
+    }
+}
+
+pub fn run(options: Options) -> anyhow::Result<()> {
+    let profile = profile::default()?;
+    let signer = term::signer(&profile)?;
+    let storage = keys::storage(&profile, signer)?;
+
+    let urn = options
+        .urn
+        .or_else(|| radicle_common::project::cwd().ok().map(|(urn, _)| urn))
+        .ok_or_else(|| anyhow!("Couldn't get URN from either command line or cwd"))?;
+
+    let identity = any::get(&storage, &urn)?
+        .ok_or_else(|| anyhow!("No project or person found for this URN"))?;
+
+    match identity {
+        SomeIdentity::Project(_) => {
+            let payload =
+                serde_json::to_string_pretty(project::verify(&storage, &urn)?.unwrap().payload())?;
```

Comment:
What is the `unwrap()` here?

Reply:
Since we've already loaded the `urn` using `any::get` once and we know it exists, we can safely call unwrap on the `Option` here.

Reply:
myeah... since you're reading from the file-system you actually can't guarantee that. It's too bad there's no `any::verify` or something like that..

Reply:
`::verify` functions return `Result<Option<VerifiedX>, Error>`.

I assumed if there's a fs failure, we'd get an `Error` from `Result`, no?

Reply:
Yeah, but if some other process removed the identity in the meantime, you'd get a panic instead of a "not found" error

Reply:
Ok, fixed https://github.com/radicle-dev/radicle-cli/commit/3ff55c8f057a810425db5a3f3c6d15ae02c9bc79#diff-7f6d0944507bd7f26b50af3f4bce4555849243552f506948e0acbbffe597f4f8R83-R84

Reply:
> myeah... since you're reading from the file-system you actually can't guarantee that. It's too bad there's no `any::verify` or something like that..

also `any::verify` would be easy to implement. what do you think @FintanH ? I can take of that.

Reply:
I took a quick look at the identity code, and I'm not sure how easy it would be to do it efficiently. I wouldn't accept a patch that tries to load `project::verify` and if that fails then try `person::verify`. But if you're willing to do dig in and find a way to get `SomeVerifiedIdentity` then I'm happy to review that :)